### PR TITLE
Refs [TASK][29160] Write and setup navigation graph

### PR DESCRIPTION
**Pull request by aug16vcc**
## Related Tickets

- [#Task_29160](hhttps://edu-redmine.sun-asterisk.vn/issues/29160?issue_count=16&issue_position=1&next_issue_id=28427)

## WHAT
- Write and setup navigation graph
## HOW
- Write and setup navigation graph
## Review Checklist

| Category      | View Point                                                   | Description                                                  | Expected Reviewer Answer | Vu Chi Cong  | Reviewer2 (name)   |
| ------------- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------ | ------------------ | ------------------ |
| General       | Code should be self-explanatory.                             | Get a feel of story reading, while going through the code. Use appropriate name for variables, functions and classes. If you are taking more time to understand the code, then either code needs refactoring or at least comments have to be written to make it clear. | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| General       | Is this function or class too big? If yes, is the function or class having too many responsibilities? |                                                              | NO                       | <li>- [x] no</li>  | <li>- [ ] no</li>  |
| General       | Does the code follow DRY?: Do not Repeat Yourself            | Is there any redundant or duplicate code? DRY (Do not Repeat Yourself) principle: The same code should not be repeated more than twice. Consider reusable services, functions and components. Consider generic functions and classes. | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| General       | Exception handling and cleanup (dispose) resources           |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| General       | Does the code build and execute correctly?                   |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Redmine       | Does the ticket follow Sun* Redmine working process?         | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |

### CODE REVIEW

File: app/src/main/java/com/sunasterisk/thooi/ui/main/MainActivity.kt [hunk 1]
```diff
@@ -2,14 +2,23 @@ package com.sunasterisk.thooi.ui.main
...
 
 class MainActivity : BaseActivity<ActivityMainBinding>() {
 
+    private val navController by lazy { findNavController(R.id.navHostFragment) }
+
     override fun onCreateBinding(
         inflater: LayoutInflater,
         container: ViewGroup?,
-        attachToRoot: Boolean
+        attachToRoot: Boolean,
```
Comment (tuanpa-0908) [inline app/src/main/java/com/sunasterisk/thooi/ui/main/MainActivity.kt:18]:
```suggestion
        attachToRoot: Boolean
```

Comment (aug16vcc) [reply app/src/main/java/com/sunasterisk/thooi/ui/main/MainActivity.kt:18]: Nope, đây là cú pháp của kotlin 1.4

File: app/src/main/res/layout/activity_main.xml [hunk 2]
```diff
@@ -6,13 +6,42 @@
...
+            android:layout_height="0dp"
+            app:defaultNavHost="true"
+            app:layout_constraintBottom_toTopOf="@id/bottomNavView"
+            app:layout_constraintTop_toTopOf="parent"
+            app:navGraph="@navigation/nav_graph"
+            tools:ignore="FragmentTagUsage" />
+
+        <com.google.android.material.bottomnavigation.BottomNavigationView
+            android:id="@+id/bottomNavView"
+            android:layout_width="match_parent"
```
Comment (tuanpa-0908) [inline app/src/main/res/layout/activity_main.xml]:
```suggestion
            android:layout_width="0dp"
```

File: app/src/main/res/menu/bottom_nav.xml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    <item
+        android:id="@+id/scheduleFragment"
+        android:contentDescription="@string/label_schedule"
+        android:icon="@drawable/ic_round_calendar_24"
+        android:title="@string/label_schedule" />
+
+    <item
+        android:enabled="false"
+        android:title="@null" />
+    <item
```
Comment (tuanpa-0908) [inline app/src/main/res/menu/bottom_nav.xml]:
```suggestion
        android:title="@null" />
        
    <item
```

File: app/src/main/res/navigation/nav_graph.xml [hunk 4]
```diff
@@ -3,15 +3,161 @@
...
+        tools:layout="@layout/fragment_sign_up">
+        <action
+            android:id="@+id/action_signupFragment_to_signInFragment"
+            app:destination="@id/signInFragment"
+            app:popUpTo="@id/signInFragment" />
+    </fragment>
+
+    <fragment
+        android:id="@+id/settingFragment"
+        android:label="Setting">
```
Comment (tuanpa-0908) [inline app/src/main/res/navigation/nav_graph.xml]: .

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -31,12 +31,17 @@
     <string name="label_schedule_manager">Quản lý công việc</string>
     <string name="label_schedule">Lịch trình</string>
     <string name="label_history">Lịch sử</string>
+    <string name="label_home">Trang chủ</string>
+    <string name="label_post_create">Tạo bài viết</string>
+    <string name="label_ranking">Xếp hạng</string>
+    <string name="label_setting">Cài đặt</string>
+
```
Comment (tuanpa-0908) [inline app/src/main/res/values/strings.xml:38]:
```suggestion
```

File: app/src/main/res/layout/activity_main.xml [hunk 6]
```diff
@@ -6,13 +6,42 @@
...
-        android:layout_height="wrap_content"
-        android:text="Hello World!"
-        app:layout_constraintBottom_toBottomOf="parent"
-        app:layout_constraintLeft_toLeftOf="parent"
-        app:layout_constraintRight_toRightOf="parent"
-        app:layout_constraintTop_toTopOf="parent" />
+        <fragment
+            android:id="@+id/navHostFragment"
+            android:name="androidx.navigation.fragment.NavHostFragment"
+            android:layout_width="match_parent"
```
Comment (tuanpa-0908) [inline app/src/main/res/layout/activity_main.xml]:
```suggestion
            android:layout_width="0dp"
```

File: app/src/main/java/com/sunasterisk/thooi/ui/main/MainActivity.kt (hunk 1 above)
Comment (tuanpa-0908) [reply app/src/main/java/com/sunasterisk/thooi/ui/main/MainActivity.kt:18]: trailing comma trong trường hợp này không cần thiết

**Pull request merged**